processing/src/main/java/io/druid/query/lookup/LookupConfig.java

```diff
@@ -26,12 +26,21 @@
 public class LookupConfig
 {
 
-  @JsonProperty
-  private final String snapshotWorkingDir;
+  @JsonProperty("snapshotWorkingDir")
+  private String snapshotWorkingDir;
+
+  @JsonProperty("enableLookupSyncOnStartup")
+  private boolean enableLookupSyncOnStartup = true;
+
+  @JsonProperty("numLookupLoadingThreads")
+  private int numLookupLoadingThreads = Runtime.getRuntime().availableProcessors() / 2;
 
   /**
    * @param snapshotWorkingDir working directory to store lookups snapshot file, passing null or empty string will disable the snapshot utility
+   * @param numLookupLoadingThreads number of threads for loading the lookups as part of the synchronization process
+   * @param enableLookupSyncOnStartup decides whether the lookup synchronization process should be enabled at startup
    */
+
   @JsonCreator
```

Comment:
I'm not sure that this configuration: `@JsonCreator` constructor with only one parameter, but two more private fields, - works. Could you please add a test, attempting to deserialize config with non-default values?